src/tokenizer/combine_fix_mark.cpp

```diff
@@ -1417,10 +1417,7 @@ void mark_function(Chunk *pc)
             LOG_FMT(LFCN, "%s(%d): orig line is %zu, orig col is %zu, function type, changing '%s' into a type\n",
                     __func__, __LINE__, pc->GetOrigLine(), pc->GetOrigCol(), pc->Text());
 
-            if (tmp2->IsNotNullChunk())
-            {
-               tmp2->SetType(CT_FUNC_TYPE);
-            }
+            tmp2->SetType(CT_FUNC_TYPE);
```

Comment:
Something is very wrong here, even in the original code. At line 1405 we check if `tmp2` is not the null chunk. If it is not, we execute the if-body at lines 1406-1414. If the if failed, we enter the else-body at lines 1416-1422. Setting tmp2 to `CT_FUNC_TYPE` is wrong, because tmp2 is a null chuck, it can't be a function. 
The original code is wrong as well, but at least the if at original line 1420 would have prevented the additional problem, because at that line tmp2 is indeed the null chuck.